File: DataGridViewScrollExample/DataGridViewScrollExample_CSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace DataGridViewScrollExample_CSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private Guna.UI.Lib.ScrollBar.DataGridViewScrollHelper hScrollHelper;
        private Guna.UI.Lib.ScrollBar.DataGridViewScrollHelper vScrollHelper;
        private void Form1_Load(object sender, EventArgs e)
        {
            hScrollHelper = new Guna.UI.Lib.ScrollBar.DataGridViewScrollHelper(DataGridView1, GunaHScrollBar1, true);
            vScrollHelper = new Guna.UI.Lib.ScrollBar.DataGridViewScrollHelper(DataGridView1, GunaVScrollBar1, true);
            Random r = new Random();
            for (int i = 0; i <= 40; i++)
            {
                DataGridView1.Rows.Add(new object[] { i, r.Next(200000, 500000), r.Next(200000, 500000), r.Next(200000, 500000), r.Next(200000, 500000) });
            }
            hScrollHelper.UpdateScrollBar();
            vScrollHelper.UpdateScrollBar();
        }

        private void DataGridView1_Resize(object sender, EventArgs e)
        {
            if (hScrollHelper != null) hScrollHelper.UpdateScrollBar();
            if (vScrollHelper != null) vScrollHelper.UpdateScrollBar();
        }
    }
}

File: DataGridViewScrollExample/DataGridViewScrollExample/Form1.vb
Public Class Form1
    Dim hScrollHelper As Guna.UI.Lib.ScrollBar.DataGridViewScrollHelper
    Dim vScrollHelper As Guna.UI.Lib.ScrollBar.DataGridViewScrollHelper

    Private Sub DataGridView1_Resize(sender As Object, e As EventArgs) Handles DataGridView1.Resize
        If hScrollHelper IsNot Nothing Then hScrollHelper.UpdateScrollBar()
        If vScrollHelper IsNot Nothing Then vScrollHelper.UpdateScrollBar()
    End Sub

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        hScrollHelper = New Guna.UI.Lib.ScrollBar.DataGridViewScrollHelper(DataGridView1, GunaHScrollBar1, True)
        vScrollHelper = New Guna.UI.Lib.ScrollBar.DataGridViewScrollHelper(DataGridView1, GunaVScrollBar1, True)
        Dim r As New Random
        For i As Integer = 0 To 40
            DataGridView1.Rows.Add({i, r.Next(200000, 500000), r.Next(200000, 500000), r.Next(200000, 500000), r.Next(200000, 500000)})
        Next
        hScrollHelper.UpdateScrollBar()
        vScrollHelper.UpdateScrollBar()
    End Sub


End Class
